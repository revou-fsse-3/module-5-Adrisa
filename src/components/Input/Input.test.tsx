import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import Input from ".";

describe("Input component unit testing", () => {
  const mocking = {
    onChange: jest.fn(),
  };

  test("renders correctly with default values", () => {
    const { asFragment } = render(
      <Input
        placeHolder="testing-placeholder"
        handleOnChange={mocking.onChange}
        value={undefined}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("renders with the correct placeholder content", () => {
    render(
      <Input
        placeHolder="testing-placeholder"
        handleOnChange={mocking.onChange}
        value={undefined}
      />
    );
    const element = screen.getByPlaceholderText("testing-placeholder");
    expect(element).toMatchSnapshot();
  });

  test("renders input with the initial value", () => {
    render(
      <Input
        placeHolder="testing-placeholder"
        handleOnChange={mocking.onChange}
        value="test value"
      />
    );

    const input = screen.getByPlaceholderText("testing-placeholder");

    expect(input).toBeDefined();
    expect(screen.getByDisplayValue("test value")).toMatchSnapshot();
  });

  test("calls onChange when input value changes", async () => {
    render(
      <Input
        placeHolder="testing-placeholder"
        handleOnChange={mocking.onChange}
      />
    );

    const input = screen.getByPlaceholderText("testing-placeholder");

    // Change the input value
    fireEvent.change(input, { target: { value: "new value" } });

    // Check if the onChange function is called once with the new value
    await waitFor(() => {
      expect(mocking.onChange).toHaveBeenCalled();
      //   expect(mocking.onChange).toHaveBeenCalledWith("new value");
    });
  });
});
