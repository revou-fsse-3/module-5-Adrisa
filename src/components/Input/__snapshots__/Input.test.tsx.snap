// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component unit testing renders correctly with default values 1`] = `
<DocumentFragment>
  <div
    class="relative"
  >
    <input
      class="block p-4 pl-10 w-full text-sm text-gray-900 bg-gray-50 rounded-lg border border-gray-300 focus:ring-blue-500 focus:border-blue-500 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
      id="default-search"
      placeholder="testing-placeholder"
      required=""
      type="search"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`Input component unit testing renders input with the initial value 1`] = `
<input
  class="block p-4 pl-10 w-full text-sm text-gray-900 bg-gray-50 rounded-lg border border-gray-300 focus:ring-blue-500 focus:border-blue-500 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
  id="default-search"
  placeholder="testing-placeholder"
  required=""
  type="search"
  value="test value"
/>
`;

exports[`Input component unit testing renders with the correct placeholder content 1`] = `
<input
  class="block p-4 pl-10 w-full text-sm text-gray-900 bg-gray-50 rounded-lg border border-gray-300 focus:ring-blue-500 focus:border-blue-500 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
  id="default-search"
  placeholder="testing-placeholder"
  required=""
  type="search"
  value=""
/>
`;
